### PR TITLE
fixes #622

### LINKED ISSUE #622: Matching Batch-Sizes and Buffer-Sizes
User do not have the option to specify Config.IN_BATCH_SIZE nor Config.OUT_BATCH_SIZE. However, in case the user specifies large Options.rcvbuf or Options.sndbuf (in out case the payload of one message is ~16MB) it could be worthwhile to adjust Config.IN_BATCH_SIZE or  Config.OUT_BATCH_SIZE accordingly in order to reduce system calls.

### PR DESCRIPTION
**Pull request by markif**
Addresses issue #622

### CODE REVIEW

File: src/main/java/zmq/io/StreamEngine.java [hunk 1]
```diff
@@ -254,14 +254,18 @@ public void plug(IOThread ioThread, SessionBase session)
         handle = ioObject.addFd(fd);
         ioError = false;
 
+        //  Make sure batch sizes match large buffer sizes
+        final int inBatchSize = options.rcvbuf > Config.IN_BATCH_SIZE.getValue() ? options.rcvbuf : Config.IN_BATCH_SIZE.getValue();
+        final int outBatchSize = options.sndbuf > Config.OUT_BATCH_SIZE.getValue() ? options.sndbuf : Config.OUT_BATCH_SIZE.getValue();
```
Comment (daveyarwood) [inline src/main/java/zmq/io/StreamEngine.java]: These could be simplified using Math.max:

```java
final int inBatchSize = Math.max(options.rcvbuf, Config.IN_BATCH_SIZE.getValue());
final int outBatchSize = Math.max(options.sndbuf, Config.OUT_BATCH_SIZE.getValue());
```

File: src/main/java/zmq/io/StreamEngine.java [hunk 2]
```diff
@@ -469,13 +473,16 @@ public void outEvent()
             outpos.set(null);
             outsize = encoder.encode(outpos, 0);
 
-            while (outsize < Config.OUT_BATCH_SIZE.getValue()) {
+            //  Make sure batch sizes match large buffer sizes
+            final int outBatchSize = options.sndbuf > Config.OUT_BATCH_SIZE.getValue() ? options.sndbuf : Config.OUT_BATCH_SIZE.getValue();
```
Comment (daveyarwood) [inline src/main/java/zmq/io/StreamEngine.java]: same deal here re: Math.max

File: src/main/java/zmq/io/StreamEngine.java [hunk 3]
```diff
@@ -607,6 +614,10 @@ private boolean handshake()
         //  Position of the version field in the greeting.
         final int revisionPos = SIGNATURE_SIZE;
 
+        //  Make sure batch sizes match large buffer sizes
+        final int inBatchSize = options.rcvbuf > Config.IN_BATCH_SIZE.getValue() ? options.rcvbuf : Config.IN_BATCH_SIZE.getValue();
+        final int outBatchSize = options.sndbuf > Config.OUT_BATCH_SIZE.getValue() ? options.sndbuf : Config.OUT_BATCH_SIZE.getValue();
```
Comment (daveyarwood) [inline src/main/java/zmq/io/StreamEngine.java]: and here as well

File: src/main/java/zmq/io/StreamEngine.java (hunk 1 above)
Comment (markif) [reply src/main/java/zmq/io/StreamEngine.java]: Seems reasonable.

**Comment by markif** [comment]
Thanks for your comments and sorry for the delay (must have slipped through the cracks). I applied the corresponding changes. I neither know the difference/history between `options` and `Config`. `Config` seems to be static (hidden from the user) whereas the user can apply changes through `options`.

**Pull request merged**